subsys/bluetooth/host/gatt.c

```diff
@@ -706,13 +728,11 @@ static ssize_t db_hash_read(struct bt_conn *conn,
 			    const struct bt_gatt_attr *attr,
 			    void *buf, uint16_t len, uint16_t offset)
 {
-	int err;
-
 	/* Check if db_hash is already pending in which case it shall be
 	 * generated immediately instead of waiting for the work to complete.
 	 */
-	err = k_delayed_work_cancel(&db_hash_work);
-	if (!err) {
+	(void)k_work_cancel_delayable_sync(&db_hash.work, &db_hash.sync);
+	if (!atomic_test_bit(gatt_sc.flags, DB_HASH_VALID)) {
```

Comment:
FWIW, atomic doesn't add much here.  AFAICT there's no lock held, so nothing prevents SMP or preemption from changing the state of the flag (and the hash) between the check and the point where changes are made assuming the check is still valid.